### DISCUSSION TITLE
Warning icon on Avatar

### DISCUSSION THREAD

**Discussion by cdpestitschek** [twilio-labs/paste#2246] [Q&A] [answered]
**Background:** This iteration is about creating a notification in User profile / settings / help center which would show/represent if something is wrong related to health status of users system.
To see more details about the user profile and avatar, [check here](https://github.com/twilio-labs/paste/discussions/2245)

**User statement**
> I as an agent would like to be aware if something is wrong with my system, having this information I would know something is wrong and i should get support.

We are looking for accessibility recommendations, including icon sizing and color.

**Stage of design:** Define/High-fidelity
**Research:** TBD
**Links to mockups or prototype:** [Figma file](https://www.figma.com/proto/UtXzwC57mVVubxGPfxZmji/Help-Centre-Supportability?node-id=154%3A33153&scaling=min-zoom&page-id=142%3A29013&starting-point-node-id=154%3A31459&show-proto-sidebar=1)

![Top Nav (1)](https://user-images.githubusercontent.com/100429369/155745244-4521b4d0-31d2-47a8-bc18-e967dd7a57bf.png)

![02b](https://user-images.githubusercontent.com/100429369/155745286-16eb1d11-f283-4c4f-9c2c-342580bee16a.png)

![02c](https://user-images.githubusercontent.com/100429369/155745312-e3fe58c1-8419-4899-98c4-b8652285dd0c.png)

**Comment by nkrantz** [comment, marked as answer]
Hi @cdpestitschek -

Because you're trying to implement a new type of flow here (see a warning > navigate to health status message > take action) it's important that we're thinking through its future potential and scalability. Showing the warning icon on the user Avatar (and the system status update in the user control center) is very specific to the user. If the only warnings that you'll ever want to show going forward are user-related warnings, that's okay, but we want to make sure we're thinking thoroughly about any future (non-user-related) warnings you might want to display. 

In order to create a scalable flow here for identification of warnings/errors, let's think of a comprehensive, future-proof solution rather than coming up with a specific solution for user errors only. Could there ever be a possibility of needing to display system-related errors that wouldn't make sense in the user control center? What if there were so many errors that needed to fit inside of the user control center, that it actually took away from the main purpose and actions of the user control center?

Thinking through these types of situations will help to be able to define scalable flows for different types and severities of warning/error. It might make sense to pull out the system health information from the user control center into a separate item in the masthead, specifically for dealing with warnings.

**Reply by shleewhite** [reply]
Hi @cdpestitschek Do you have what you need or would you like more assistance with this?

**Reply by cdpestitschek** [reply]
Not for now @shleewhite. @nkrantz thank you so much for the feedback && clarification.

**Comment by bchicca** [comment]
@nkrantz I have a different use case for showing a specific error about the user profile not loading. 
Is this combination allowed: error icon + popover?
![Screenshot 2023-05-25 at 16 48 31](https://github.com/twilio-labs/paste/assets/114075970/8c4b842f-e899-45f4-90f1-a9fd0b99e1c7)

**Reply by TheSisb** [reply]
Hi @bchicca,

From a feasibility perspective, you are able to [trigger a Popover on an icon-only button](https://paste.twilio.design/components/popover#popoverbutton).

From a design perspective, it's hard to say.  
- If this message pertains to the page that is being loaded, the message should be presented on the page using the [Alert component.](https://paste.twilio.design/components/alert)
- If this is a system wide-warning, you may want to consider Nora's answer from above. 

Hope this helps

**Comment by bchicca** [comment]
@TheSisb this is an error pertaining the profile component. When there is a problem with that component, users are not able to log out. It's tricky because it's a component error, not a system error, not a page error, and the component is very small to display a help text.
I haven't seen this use case covered in the error patterns page, is it something that would make sense adding?
